lib/Terrain.cpp

```diff
@@ -440,6 +440,16 @@ bool TerrainType::isUnderground() const
 	return passabilityType & PassabilityType::SUBTERRANEAN;
 }
 
+bool TerrainType::isSurfaceCartographerCompatible() const
+{
+	return isLand() && id != Terrain::SUBTERRANEAN && id != Terrain::ROCK;
+}
+
+bool TerrainType::isUndergroundCartographerCompatible() const
+{
+	return isLand() && id == Terrain::SUBTERRANEAN;
```

Comment:
1) What about ROCK? Not 100% sure but I suspect that it should be "revealed" as well by this cartographer - to replace stars from terra incognita with revealed unpassable rocks

2) Don't use terrain types directly, instead use traits from config - we may have new terrains added by mods.
I see isPassable() check that can replace ROCK and isUnderground() that can replace direct check for SUBTERRANEAN

Reply:
ad. 1: ROCK is not revealed in original H3

ad. 2: isUnderground() also includes lava in our config

Reply:
Hmm... Interesting, so there is no way to reveal ROCK in H3?
Okay then.

2 - In that case use isSurface()